hexa/databases/utils.py

```diff
@@ -0,0 +1,120 @@
+from typing import Dict, List, Tuple
+
+import psycopg2
+from psycopg2 import sql
+
+from hexa.core.graphql import result_page
+from hexa.plugins.connector_postgresql.models import Database
+from hexa.workspaces.models import Workspace
+
+
+def format_sample_data(columns, samples):
+    res = []
+    for sample in samples:
+        res.append(
+            [{"column": columns[j], "value": sample[j]} for j in range(len(sample))]
+        )
+    return res
+
+
+def get_database_definition(
+    workspace: Workspace, limit_per_table=4, page=1, per_page=10
+):
+    tables_summary = []
+    IGNORE_TABLES = ["geography_columns", "geometry_columns", "spatial_ref_sys"]
+
+    database = Database.objects.get(database="hexa-explore-demo")
+    url = database.url
+    # url = workspace.database.url
+    with psycopg2.connect(url) as conn:
+        with conn.cursor(cursor_factory=psycopg2.extras.DictCursor) as cursor:
+            cursor.execute(
+                """
+                    SELECT table_name, pg_class.reltuples as row_count
+                    FROM information_schema.tables
+                    JOIN pg_class ON information_schema.tables.table_name = pg_class.relname
+                    WHERE table_schema = 'public'
+                    ORDER BY table_name;
+                """
+            )
+            response: List[Tuple[str, str, int]] = cursor.fetchall()
+            tables: Dict[str, Dict] = {
+                x[0]: x for x in response if x[0] not in IGNORE_TABLES
+            }
+
+            for name, data in tables.items():
+                row_count = data["row_count"]
+                if data["row_count"] < 10_000:
```

Comment:
This part can be removed. The estimate is good enough plus querying a count for each table separately is not really a good idea as it will potentially trigger 1 query per table

Reply:
Yes, I agree.